### PR TITLE
[TN] raise NotImplementedError for unsupported languages and other minor fixes

### PR DESCRIPTION
**Pull request by XuesongYang**
Signed-off-by: Xuesong Yang <1646669+XuesongYang@users.noreply.github.com>

# What does this PR do ?

* raised `NotImplementedError` for unsupported languages.
* added `@staticmethod` decor for member functions.
* fixed some typos.
* renamed argument name for the subroutine of `generate_permutations()` to avoid inner and outer scopes.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo_text_processing/text_normalization/normalize.py [hunk 1]
```diff
@@ -406,12 +407,13 @@ def find_tags(self, text: str) -> 'pynini.FstLike':
         lattice = text @ self.tagger.fst
         return lattice
 
-    def select_tag(self, lattice: 'pynini.FstLike') -> str:
+    @staticmethod
+    def select_tag(lattice: 'pynini.FstLike') -> str:
```
Comment (yzhang123) [inline nemo_text_processing/text_normalization/normalize.py:411]: is this function used anywhere else? since it's now a static method

**Comment by XuesongYang** [comment]
@yzhang123 please let me know if any other changes are needed. Thanks.

File: nemo_text_processing/text_normalization/normalize.py [hunk 2]
```diff
@@ -70,6 +56,19 @@ def __init__(
         lm: bool = False,
         post_process: bool = True,
     ):
+        """
```
[the remark this answers is not in the document]
Comment (XuesongYang) [reply nemo_text_processing/text_normalization/normalize.py:59]: i saw different conventions across NeMo domains, but TN used consistent ones. I will revert.

**Comment by XuesongYang** [comment]
@yzhang123 @ekmb is there anything to add for this minor fix?

**Review by ekmb — APPROVED** [review]
LGTM, thanks!

**Pull request merged**